java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -182,10 +182,13 @@ public static long estimateRgEndOffset(boolean isCompressed,
                                          long streamLength) {
     // figure out the worst case last location
     // if adjacent groups have the same compressed block offset then stretch the slop
-    // by factor of 2 to safely accommodate the next compression block.
-    // One for the current compression block and another for the next compression block.
+    // by a factor to safely accommodate the next compression block.
+    // 512 is the MAX_SCOPE defined in RunLengthIntegerWriterV2.
```

Comment:
Btw, we should probably expose and use MAX_SCOPE and MAX_SIZE directly from RunLengthIntegerWriterV2 instead of relying on plain values